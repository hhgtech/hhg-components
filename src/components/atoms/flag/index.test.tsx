import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Flag } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const flag = render(
    <MockTheme>
      <Flag country="ID" width={100} height={100} />
    </MockTheme>,
  )

  const self = flag.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toHaveAttribute('data-is-bitmap', 'true')
})
