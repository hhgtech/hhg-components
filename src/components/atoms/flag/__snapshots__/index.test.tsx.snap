// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  --width: 100;
  --height: 100;
}

.c0 img,
.c0 svg {
  width: calc(var(--width) * 1px);
  height: calc(var(--height) * 1px);
}

.c0[data-is-bitmap] img {
  object-fit: contain;
}

<div
  class="c0"
  data-is-bitmap="true"
>
  <img
    height="100"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAhCAYAAACfiCi5AAAACXBIWXMAABYlAAAWJQFJUiTwAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAACrSURBVHgB7dY9CgIxEIbhb8JqnV6QNILltnbuUTyRR/AK3kBrm/UIOUJKwZ9x0tsPn8wDCSFVXkggQPAlfbrlzbgQHBWyBwGBXh/6POxarTLnbVH5zLafwaW9VKcEvE/gO3yXB7s1SYXj2vwmYwK3zB6ACPAWAd4iwFsEeKMPGJbrFZiJGhCLN+AtArxFgLcI8BYB3v4ioIFX6wF38Dr332ixxcVGAZdqY/oC5+Yj9mTCW5wAAAAASUVORK5CYII="
    width="100"
  />
</div>
`;
